data-processing/entities/citations/types.py

```diff
@@ -26,3 +26,4 @@ class BibitemMatch:
     bibitem_text: str
     s2_id: str
     s2_title: str
+    similarity_score: float
```

Comment:
Curious about why we're adding this here?

Reply:
It will help us in future debugging. Without we have no way of getting the score in case we need to look at the similarity match score.